### PR TITLE
Add control variables to Jenkins E2E script

### PR DESCRIPTION
**Pull request by saad-ali**
This will enable me finer grained control to create Jenkins jobs that only bring a cluster up, or only run tests, etc. (needed for setting up soak test). Default for all new variable is enabled, so existing jobs will continue to work without modification.

CC @quinton-hoole

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -92,6 +92,11 @@ export KUBE_GKE_NETWORK=${E2E_NETWORK}
 export PATH=${PATH}:/usr/local/go/bin
 export KUBE_SKIP_CONFIRMATIONS=y
 
+# E2E Control Variables
+export E2E_UP="${E2E_UP:-true}"
+export E2E_TEST="${E2E_UP:-true}"
```
Comment (quinton-hoole) [inline hack/jenkins/e2e.sh]: Cut 'n paste error?

**Comment by quinton-hoole** [comment]
It's a bit clunky to have E2E_UP mean "--down, --up, --version" under the hood, but there's no better, simpler way that I can think of right now.  LGTM other than the apparent cut'n paste error which needs to be fixed.

**Comment by saad-ali** [comment]
Good catch, fixed.

**Comment by quinton-hoole** [comment]
Travis and shippable are not going to build or test this anyway, so merging to see whether Jenkins still works.

**Pull request merged**